hc/front/views.py

```diff
@@ -27,6 +32,107 @@ def pairwise(iterable):
     return zip(a, b)
 
 
+def blogs(request, filter_by):
+    num = 3
+    category = Category.objects.all()
+    if int(filter_by) > 0:
+        category_id = Category.objects.get(pk=filter_by).id
+        stories = Blog_post.objects.filter(category=category_id)
+    else:
+        stories = Blog_post.objects.all()
+    list_of_stories = [story for story in stories]
+    blogs = [list_of_stories[i:i+num] for i in range(0, len(list_of_stories), num)]
```

Comment:
Nice comprehension here. However, add a comment about what you want to achieve for code easy code maintainability.